# INITIAL DATA EXPLORATION

> What is conducted in this notebook?
1. Shape of the dataset is studied, and the columns are seen in details. 
2. Data description and Summary Statistics
3. Normalization of data to support EDA analysis
4. Checking for NULL and Duplicates
5. EDA 
6. Dropping of columns which are not necessary for model building
7. Saving final dataframe to csv.

In [45]:
#IMPORTING LIBRARIES
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set()
import warnings
warnings.simplefilter("ignore")

## LODING DATASET 

In [46]:

file_path = r"C:\Data_Science\Mini Project\Capston Project\Capston_IOD_Project\DATABASE files\amazon_reviews_2019.csv"

# Try loading the CSV file with a different encoding
try:
    data = pd.read_csv(file_path, encoding='ISO-8859-1')  # or use 'cp1252'
    #REMOVE MAX
    pd.set_option('display.max_columns', None)
    pd.set_option('display.max_rows', None)
    
except Exception as e:
    print("Error loading file:", e)


## SHAPE OF THE DATA AND ITS ATTRIBUTES

In [47]:
#TOTAL ROWS AND COLUMNS
n_reviews, n_attributes = data.shape
print('Total Reviews: ',n_reviews)
print('Total Attributes: ',n_attributes)

Total Reviews:  2501
Total Attributes:  32


> There are in total of 2.5k reviews within this dataset, which makes it to be a valid dataset since the records are more than 1k, thus suitable for analysis.

> However, it is to be noted that there are in total of 32 columns within this dataset, which is far more than what is required for this project. These columns need to be taken a deeper look for feature selection.

In [48]:
#INFORMATION ON THE LOADED CSV FILE. COVERS COLUMN NAMES, TOTAL ROWS AND COLUMS, DATA TYPES AND THE MEMORY USAGE.
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2501 entries, 0 to 2500
Data columns (total 32 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   report_date             2501 non-null   object 
 1   online_store            2501 non-null   object 
 2   upc                     2501 non-null   float64
 3   retailer_product_code   2501 non-null   object 
 4   brand                   2501 non-null   object 
 5   category                2501 non-null   object 
 6   sub_category            2501 non-null   object 
 7   product_description     2501 non-null   object 
 8   review_date             2501 non-null   object 
 9   review_rating           2501 non-null   int64  
 10  review_title            2396 non-null   object 
 11  review_text             2501 non-null   object 
 12  is_competitor           2501 non-null   int64  
 13  manufacturer            2501 non-null   object 
 14  market                  2501 non-null   

In [49]:
#DISPLAY DATAFRAME
data.head()

report_date online_store           upc retailer_product_code          brand  \
0   1/02/2019  FRESHAMAZON  8.720000e+12            B0142CI6FC  Dove Men+Care   
1   1/03/2019  FRESHAMAZON  5.000000e+12            B014DFNNRY        Marmite   
2   1/03/2019  FRESHAMAZON  5.000000e+12            B014DFNNRY        Marmite   
3   1/03/2019  FRESHAMAZON  8.710000e+12            B014DFKELC          Knorr   
4   1/03/2019  FRESHAMAZON  8.720000e+12            B014G37I7E            Cif   

        category sub_category  \
0  Personal Care         Deos   
1          Foods      Savoury   
2          Foods      Savoury   
3          Foods      Savoury   
4       Homecare          HHC   

                                 product_description review_date  \
0  Dove Men+Care Extra Fresh Anti-perspirant Deod...   1/01/2019   
1                 Marmite  Spread Yeast Extract 500g   1/02/2019   
2                 Marmite  Spread Yeast Extract 500g   1/02/2019   
3                      Knorr  Beef Stock Pot 8 x 28g   1/02/2019   
4                   Cif Citrus Bathroom Mousse 500ml   1/02/2019   

   review_rating                                       review_title  \
0              5                             Dove Men’s + Deodorant   
1              5                         Great for a marmite lover!   
2              4                                       Vitamin B12.   
3              5  A Very Handy Flavour Top - Up to Keep In The C...   
4              5                                         Very handy   

                                         review_text  is_competitor  \
0  As you get older, you know what you like and w...              0   
1  Three gigantic marmite jars that will last pro...              0   
2                                          Excellent              0   
3         A great flavour top - up for slow cooking.              0   
4                          Does what is says it does              0   

      manufacturer market  matched_keywords  time_of_publication  \
0  Unilever Global     UK               NaN                  NaN   
1  Unilever Global     UK               NaN                  NaN   
2  Unilever Global     UK               NaN                  NaN   
3  Unilever Global     UK               NaN                  NaN   
4  Unilever Global     UK               NaN                  NaN   

                                                 url review_type  \
0  http://www.amazon.co.uk/gp/customer-reviews/RE...     Organic   
1  http://www.amazon.co.uk/gp/customer-reviews/R1...     Organic   
2  http://www.amazon.co.uk/gp/customer-reviews/RD...     Organic   
3  http://www.amazon.co.uk/gp/customer-reviews/R2...     Organic   
4  http://www.amazon.co.uk/gp/customer-reviews/RJ...     Organic   

  parent_review  manufacturers_response dimension1  \
0         Child                     NaN       Deos   
1         Child                     NaN    Savoury   
2         Child                     NaN    Savoury   
3        Parent                     NaN    Savoury   
4        Parent                     NaN        HHC   

                                          dimension2  \
0                     Male Anti-Perspirant Deodorant   
1                                 COTC Yeast Extract   
2                                 COTC Yeast Extract   
3  Beef Stock/Pots/Cubes/Extract/Liquid/Concentrated   
4                                    Bathroom Mousse   

                            dimension3  dimension4  dimension5  dimension6  \
0  COTC Male Anti-Perspirant Deodorant         NaN         NaN         NaN   
1                   COTC Yeast Extract         NaN         NaN         NaN   
2                   COTC Yeast Extract         NaN         NaN         NaN   
3                           Stock Pots         NaN         NaN         NaN   
4                      Bathroom Mousse         NaN         NaN         NaN   

                            dimension7        dimension8  verified_purchase  \
0  COTC Male Anti-Perspir

## DATA DESCRIPTION


> We can gain a better grasp of the dataset by digging deeper into the columns. The columns are described in detail below.

1. report_date: Date when the data was extracted.
                Most records seem to be from 2019.
2. online_store: The name of the online store where the reviews were posted.
3. upc (Universal Product Code):A 12-digit numeric barcode used to identify trade items globally.
4. retailer_product_code:  Product identifier from the retailer's side.
5. brand: The brand of the product being reviewed.
6. category: General product category (e.g., food, electronics).
7. sub_category: A more specific product category under the general category (e.g., savoury under food).
8. product_description: A detailed description of the product.
9. review_date: The date when the review was posted.
10. review_rating: ating provided by the reviewer, on a scale from 1 (very poor) to 5 (excellent).
11. review_title: The title of the review.
12. review_text: The full text of the review, often containing feedback on the product.
13. is_competitor: Whether the product is considered a competitor's product. Requires further analysis for clarity.
14. manufacturer: the manufacturer of the product.
15. market:The geographical location or market where the products and stores are situated.
16. matched_keywords: No data available (NULL values).
17. time_of_publication: No data available (NULL values).
18. url: URL link to the original review.
19. review_type: Indicates the type of review. This column requires additional investigation for clarity.
20. parent_review: Indicates whether the review is a parent or child review in a discussion thread.
21. manufacturers_response:Response from the manufacturer to the review, if any.
22. dimension 1 - 8:Miscellaneous product-related data, but several missing values exist, which need verification.
23. verified_purchase: Indicates if the review is from a verified purchase.
24. helpful_review_count: The number of individuals who found the review helpful.
25. review_hash_id: A unique identifier for each review.

## SUMMARY STATISTICS 

> Summary statistics help us understand the dataset's overall structure. For numerical columns, they provide insights into max/min values, averages, and the presence of null values. For categorical columns, they reveal unique values and their frequencies, helping us identify patterns and potential issues like missing data.

In [50]:
#NON-OBJECTS
data.describe()

upc  review_rating  is_competitor  matched_keywords  \
count  2.501000e+03    2501.000000         2501.0               0.0   
mean   7.631221e+12       4.456218            0.0               NaN   
std    2.108136e+12       1.108595            0.0               NaN   
min    4.218266e+07       1.000000            0.0               NaN   
25%    8.710000e+12       4.000000            0.0               NaN   
50%    8.710000e+12       5.000000            0.0               NaN   
75%    8.710000e+12       5.000000            0.0               NaN   
max    8.720000e+12       5.000000            0.0               NaN   

       time_of_publication  manufacturers_response  dimension4  dimension5  \
count                  0.0                     0.0         0.0         0.0   
mean                   NaN                     NaN         NaN         NaN   
std                    NaN                     NaN         NaN         NaN   
min                    NaN                     NaN         NaN         NaN   
25%                    NaN                     NaN         NaN         NaN   
50%                    NaN                     NaN         NaN         NaN   
75%                    NaN                     NaN         NaN         NaN   
max                    NaN                     NaN         NaN         NaN   

       dimension6  helpful_review_count  
count         0.0           2501.000000  
mean          NaN              0.231507  
std           NaN              0.953930  
min           NaN              0.000000  
25%           NaN              0.000000  
50%           NaN              0.000000  
75%           NaN              0.000000  
max           NaN              7.000000

1. Average Rating: 4.46 (positive reviews).
2. Helpful Count: Average is 0.23 (most reviews not marked as helpful).
3. UPC: Unique identifier, not relevant for analysis.
4. Competitor Flag: All values are 0 (no competitors flagged).
5. .Null Columns: matched_keywords, time_of_publication, manufacturers_response, and dimension4-6 are entirely NULL.




In [51]:
#OBJECTS
data.describe(include=object)

report_date online_store retailer_product_code    brand       category  \
count         2501         2501                  2501     2501           2501   
unique         133            9                   367       35              4   
top      5/26/2019       AMAZON            B077YLQ2R1  Comfort  Personal Care   
freq           217         1102                    97      480           1182   

       sub_category                                product_description  \
count          2501                                               2501   
unique           16                                                219   
top         Laundry  Simple x Little Mix Micellar Cleansing Wipes 2...   
freq            993                                                108   

       review_date review_title review_text     manufacturer market  \
count         2501         2396        2501             2501   2501   
unique         144         1243        1622                1      1   
top      2/19/2019  Great value        Good  Unilever Global     UK   
freq            70           48          25             2501   2501   

                                                      url review_type  \
count                                                1654        2501   
unique                                               1571           2   
top     https://www.sainsburys.co.uk/shop/gb/groceries...     Organic   
freq                                                   52        1936   

       parent_review dimension1               dimension2  \
count           2501       2501                     2501   
unique             2         16                      106   
top           Parent    Laundry  COTC Fabric Conditioner   
freq            1800        993                      326   

                     dimension3                 dimension7        dimension8  \
count                      2310                       2499              2501   
unique                       94                         15                 1   
top     COTC Fabric Conditioner  Retailer Core of the Core  Core of the Core   
freq                        276                       1281              2501   

                              review_hash_id  
count                                   2501  
unique                                  2501  
top     3f129b02-ea76-0323-bd59-235d97a4f83f  
freq                                       1

1. Stores: There are 9 stores, with most reviews coming from Amazon. All stores are major outlets in the UK.
2. Category: Most data is from the Personal Care category, specifically under Laundry products from the Comfort brand. 
3. Positive Reviews: Common phrases in positive reviews include "Good" and "great value," indicating similar wording in positive feedback.
4. Manufacturer: All products are from a single manufacturer, Unilever Global.
5. Market: The dataset is focused entirely on the UK.
6. Review Types: There are two review types, mostly organic and parent reviews.
7. Dimensions: Dimensions 1-8 provide additional product information, mostly related to laundry items.
8. Review Hash ID: This can be ignored as it's a unique identifier.

In [52]:
data["review_type"].value_counts(normalize=True).to_frame()

proportion
review_type            
Organic         0.77409
Syndicated      0.22591


1. Organic Reviews: Comprise 77% of the dataset, indicating a strong base of genuine customer feedback.
2. Syndicated Reviews: Account for only 22%, representing reviews shared across multiple platforms for better accessibility.

In [53]:
data["parent_review"].value_counts(normalize=True).to_frame()

proportion
parent_review            
Parent           0.719712
Child            0.280288

Most of the reviews are said to be parent reviews. Upon further research, it is unclear what this stands for, and hence might be dropped due to is ambiguity.

In [54]:
data["review_date"].value_counts().to_frame()

count
review_date       
2/19/2019       70
1/26/2019       49
2/01/2019       47
2/20/2019       46
5/01/2019       40
2/26/2019       39
1/31/2019       38
2/18/2019       36
1/14/2019       36
1/27/2019       35
2/08/2019       35
3/01/2019       34
2/03/2019       32
2/04/2019       31
2/05/2019       30
2/22/2019       29
1/28/2019       29
3/31/2019       29
2/02/2019       28
1/30/2019       28
4/02/2019       28
1/12/2019       27
1/02/2019       27
2/11/2019       26
1/25/2019       25
1/04/2019       25
1/11/2019       25
1/05/2019       25
2/06/2019       24
2/17/2019       24
3/23/2019       24
2/16/2019       24
1/16/2019       24
3/12/2019       23
1/22/2019       23
2/09/2019       22
4/07/2019       22
4/30/2019       22
1/15/2019       22
2/07/2019       22
1/20/2019       21
1/08/2019       21
1/03/2019       20
5/19/2019       19
4/01/2019       19
1/09/2019       19
4/16/2019       19
5/17/2019       19
2/12/2019       19
2/14/2019       19
1/10/2019       19
5/04/2019       18
1/06/2019       18
5/20/2019       18
5/15/2019       18
4/27/2019       18
1/19/2019       18
4/20/2019       18
5/08/2019       18
4/08/2019       18
4/03/2019       17
3/05/2019       17
3/30/2019       17
1/18/2019       17
5/09/2019       17
2/21/2019       16
5/21/2019       16
2/15/2019       16
5/06/2019       16
2/23/2019       15
4/26/2019       15
1/21/2019       15
4/18/2019       14
4/11/2019       14
4/25/2019       14
5/14/2019       14
5/07/2019       14
1/01/2019       14
3/16/2019       14
1/29/2019       14
2/24/2019       14
2/10/2019       14
3/06/2019       14
4/24/2019       13
3/07/2019       13
4/23/2019       13
2/13/2019       13
4/09/2019       13
1/07/2019       13
4/10/2019       13
5/13/2019       13
4/21/2019       12
4/29/2019       12
5/02/2019       12
4/17/2019       12
4/14/2019       12
4/12/2019       12
5/22/2019       12
2/27/2019       11
4/06/2019       11
5/11/2019       11
1/23/2019       11
4/28/2019       11
5/12/2019       11
3/02/2019       10
5/16/2019       10
3/26/2019       10
1/13/2019       10
2/25/2019       10
5/18/2019       10
3/04/2019       10
5/05/2019       10
1/24/2019        9
5/03/2019        9
3/21/2019        9
3/22/2019        9
3/11/2019        9
3/27/2019        9
3/03/2019        8
3/08/2019        8
4/15/2019        8
3/10/2019        8
3/19/2019        8
3/28/2019        8
3/15/2019        8
3/13/2019        8
3/17/2019        8
4/22/2019        7
2/28/2019        7
4/04/2019        7
4/05/2019        7
3/09/2019        7
3/14/2019        6
5/24/2019        6
3/20/2019        6
5/10/2019        5
3/24/2019        5
4/19/2019        5
5/23/2019        5
4/13/2019        4
3/25/2019        4
1/17/2019        4
3/18/2019        4
3/29/2019        3

In [55]:
data["report_date"].value_counts().to_frame()

count
report_date       
5/26/2019      217
2/28/2019      201
5/27/2019      134
3/25/2019      122
3/21/2019       81
2/22/2019       56
5/23/2019       55
3/22/2019       50
2/08/2019       39
3/24/2019       37
2/25/2019       34
2/20/2019       34
2/15/2019       33
5/06/2019       32
2/13/2019       31
3/20/2019       31
5/02/2019       30
1/31/2019       30
2/26/2019       28
3/19/2019       28
3/08/2019       28
4/30/2019       28
3/11/2019       28
2/11/2019       28
2/10/2019       24
2/04/2019       24
2/14/2019       24
5/24/2019       24
2/03/2019       23
3/10/2019       23
3/28/2019       23
2/27/2019       22
4/22/2019       22
5/03/2019       21
3/18/2019       21
2/19/2019       21
3/31/2019       21
2/07/2019       20
3/14/2019       20
5/16/2019       20
2/09/2019       19
5/10/2019       19
5/01/2019       19
4/29/2019       19
2/02/2019       18
2/18/2019       18
3/16/2019       17
4/05/2019       17
5/20/2019       17
2/24/2019       17
2/05/2019       16
2/17/2019       16
2/23/2019       15
5/09/2019       15
3/09/2019       14
5/11/2019       14
3/17/2019       14
5/17/2019       13
5/18/2019       13
2/21/2019       13
3/23/2019       13
2/06/2019       13
5/14/2019       12
5/25/2019       12
5/21/2019       12
4/20/2019       12
2/12/2019       11
3/15/2019       11
5/05/2019       10
5/07/2019       10
4/25/2019       10
4/03/2019       10
5/12/2019       10
3/05/2019       10
3/29/2019        9
2/01/2019        9
3/27/2019        8
3/13/2019        8
3/12/2019        8
4/15/2019        8
4/27/2019        8
4/26/2019        8
3/06/2019        8
3/07/2019        8
5/15/2019        8
5/08/2019        7
5/04/2019        7
4/23/2019        7
3/26/2019        7
4/11/2019        7
4/06/2019        7
5/22/2019        6
4/19/2019        6
4/10/2019        6
3/30/2019        6
5/19/2019        5
4/04/2019        5
5/13/2019        5
4/21/2019        5
1/23/2019        4
1/03/2019        4
4/01/2019        4
4/13/2019        4
4/14/2019        4
4/28/2019        4
4/16/2019        4
4/02/2019        4
4/08/2019        4
4/24/2019        3
1/12/2019        3
1/04/2019        3
1/29/2019        3
1/14/2019        2
1/07/2019        2
4/12/2019        2
2/16/2019        2
4/07/2019        2
1/28/2019        2
4/18/2019        2
4/17/2019        2
1/17/2019        2
1/06/2019        1
1/13/2019        1
4/09/2019        1
1/08/2019        1
1/10/2019        1
1/30/2019        1
1/27/2019        1
1/26/2019        1
1/22/2019        1
1/20/2019        1
1/15/2019        1
1/02/2019        1

The years of the reviews  on this dataset consists of 2019 only.

In [56]:
data.groupby('category')['sub_category'].value_counts().sort_index(ascending=True).to_frame()

count
category      sub_category                        
Foods         Dressings                         36
              Ice Cream                         39
              Other Foods                       16
              Savoury                          100
Homecare      HHC                               27
              Household Care                    30
              Laundry                          993
Personal Care Deodorants & Fragrances          125
              Deos                             100
              Hair                             108
              Hair Care                         77
              Personal Care                     22
              Skin Care                        476
              Skin Cleansing                   274
Refreshment   Ice Cream                         70
              Tea                                3
              Tea and Soy & Fruit Beverages      5

The sub-categories, grouped by the categories accordingly. 

> Standardizing: Similar sub-categories like "Deos" and "Deodorants & Fragrances," "Hair" and "Hair Care," and "Ice Cream" across different categories are merged. 

> Grouped Summary: If you need, you can group the data based on category and sub_category to sum up the counts and avoid duplication.

needs to be handled since this can be useful for further EDA analysis

In [57]:
#CHANGING THE CATEGORIES AND SUB-CATEGORIES
data.loc[data.sub_category == "Ice Cream", "category"] = "Refreshment"

In [58]:
data.loc[data.sub_category == "HHC", "sub_category"] = "Household Care"

In [59]:
data.loc[data.sub_category == "Deos", "sub_category"] = "Deodorants & Fragrances"

In [60]:
data.loc[data.sub_category == "Tea", "sub_category"] = "Tea and Soy & Fruit Beverages"

In [61]:
data.loc[data.sub_category == "Hair Care", "sub_category"] = "Hair"

In [62]:
data.groupby('category')['sub_category'].value_counts().sort_index(ascending=True).to_frame()

count
category      sub_category                        
Foods         Dressings                         36
              Other Foods                       16
              Savoury                          100
Homecare      Household Care                    57
              Laundry                          993
Personal Care Deodorants & Fragrances          225
              Hair                             185
              Personal Care                     22
              Skin Care                        476
              Skin Cleansing                   274
Refreshment   Ice Cream                        109
              Tea and Soy & Fruit Beverages      8

Everything is now sorted and hence can be used for EDA analysis accordingly.

In [63]:
data["online_store"].value_counts().to_frame()

count
online_store            
AMAZON              1102
AMAZONPRIMEPANTRY    389
BOOTS                378
SUPERDRUG            188
OCADO                168
SAINSBURYS           123
ASDA                 109
FRESHAMAZON           40
MORRISONS              4

The majority of the reviews in this dataset come from the AMAZON store, and most stores are located in the UK.

>Helpful Reviews: A large portion of the reviews hasn't received any helpful votes, with those that do being awarded very few. This could potentially skew the overall insights, making it difficult to identify fake reviews effectively based solely on helpful review counts.

## CHECKING FOR NULL AND DUPLICATED VALUES

In [64]:
#CHECKING THE NUMBER OF NULL VALUES ACROSS THE DATASET
data.isnull().sum()

report_date                  0
online_store                 0
upc                          0
retailer_product_code        0
brand                        0
category                     0
sub_category                 0
product_description          0
review_date                  0
review_rating                0
review_title               105
review_text                  0
is_competitor                0
manufacturer                 0
market                       0
matched_keywords          2501
time_of_publication       2501
url                        847
review_type                  0
parent_review                0
manufacturers_response    2501
dimension1                   0
dimension2                   0
dimension3                 191
dimension4                2501
dimension5                2501
dimension6                2501
dimension7                   2
dimension8                   0
verified_purchase            0
helpful_review_count         0
review_hash_id               0
dtype: i


The dataset contains several NULL values that need to be addressed:

1. review_title has 105 NULL values, a small minority compared to over 2,000 records.
2. URL has 800+ missing values, but can be ignored as it's not essential to the project.
3.matched_keywords, time_of_publication, manufacturers_response, dimension4, dimension5, and dimension6 contain no values and will be dropped entirely
4. dimension7 has only 2 missing values and provides extra product information, so it can be evaluated based on its importance.

**Key columns for analysis—review_rating, review_text, and verified_purchase—have no missing values, which is ideal for the project’s focus.**

In [65]:
#DISPLAYS THE NUMBER OF DUPLICATED VALUES
dup = data.duplicated().sum()
print("Number of duplicates in dataset: ", dup)

Number of duplicates in dataset:  0


### TEXTUAL ATTRIBUTES

In [66]:
#TEXTUAL ATTRIBUTES
cols = ['review_date','review_title', 'review_text','review_rating','verified_purchase']
textual_attributes = data[cols]
pd.DataFrame(textual_attributes).head()

review_date                                       review_title  \
0   1/01/2019                             Dove Men’s + Deodorant   
1   1/02/2019                         Great for a marmite lover!   
2   1/02/2019                                       Vitamin B12.   
3   1/02/2019  A Very Handy Flavour Top - Up to Keep In The C...   
4   1/02/2019                                         Very handy   

                                         review_text  review_rating  \
0  As you get older, you know what you like and w...              5   
1  Three gigantic marmite jars that will last pro...              5   
2                                          Excellent              4   
3         A great flavour top - up for slow cooking.              5   
4                          Does what is says it does              5   

   verified_purchase  
0               True  
1               True  
2               True  
3               True  
4              False


> The dataset includes textual attributes that are complete (no NULL values). These attributes can be leveraged to analyze the textual context of the reviews. By doing so, we can gain deeper insights into the review data, which will prove useful when building a classifier. 

> By focusing on linguistic features of the review text, we can better approach the problem of identifying fake vs genuine reviews

> The textual attributes will be integral to the EDA process, helping uncover patterns in the language used in reviews, which can contribute to more accurate model building.

In [67]:
#DROP AFOREMENTIONED COLUMNS
data.drop(['matched_keywords', 'time_of_publication'
         ,'manufacturers_response', 'dimension4',
         'dimension5', 'dimension6', 
         'is_competitor', 'helpful_review_count',
        'report_date','online_store','brand',
         'category','sub_category', 'market',
        'upc', 'retailer_product_code', 'review_hash_id', 
         'url','product_description', 'parent_review', 
         'review_type', 'manufacturer', 'dimension1', 
         'dimension2','dimension3','dimension4',
         'dimension5', 'dimension7', 
         'dimension8'], axis=1, inplace=True)
data.head() #UPDATED DATAFRAME

review_date  review_rating  \
0   1/01/2019              5   
1   1/02/2019              5   
2   1/02/2019              4   
3   1/02/2019              5   
4   1/02/2019              5   

                                        review_title  \
0                             Dove Men’s + Deodorant   
1                         Great for a marmite lover!   
2                                       Vitamin B12.   
3  A Very Handy Flavour Top - Up to Keep In The C...   
4                                         Very handy   

                                         review_text  verified_purchase  
0  As you get older, you know what you like and w...               True  
1  Three gigantic marmite jars that will last pro...               True  
2                                          Excellent               True  
3         A great flavour top - up for slow cooking.               True  
4                          Does what is says it does              False

In [68]:
#DROPPING UNWANTED COLUMN
# data.drop(['Unnamed: 0'], axis=1, inplace=True)

In [69]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2501 entries, 0 to 2500
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   review_date        2501 non-null   object
 1   review_rating      2501 non-null   int64 
 2   review_title       2396 non-null   object
 3   review_text        2501 non-null   object
 4   verified_purchase  2501 non-null   bool  
dtypes: bool(1), int64(1), object(3)
memory usage: 80.7+ KB


## RE-CHECK NULL AND DUPLICATES

In [70]:
#CHECK TOTAL DUPLICATE OCCURENCES
dup = data.duplicated().sum()
print("Number of duplicates in dataset: ", dup)

Number of duplicates in dataset:  783


In [71]:
data = data.drop_duplicates().reset_index(drop=True)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1718 entries, 0 to 1717
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   review_date        1718 non-null   object
 1   review_rating      1718 non-null   int64 
 2   review_title       1650 non-null   object
 3   review_text        1718 non-null   object
 4   verified_purchase  1718 non-null   bool  
dtypes: bool(1), int64(1), object(3)
memory usage: 55.5+ KB


### NULL Values
re-checking for NULL values to check if any needs to be filled up or dropped.

In [72]:
data.isnull().sum()

review_date           0
review_rating         0
review_title         68
review_text           0
verified_purchase     0
dtype: int64

In [73]:
#DROP review_title
data.drop(["review_title","review_date"], axis=1, 
        inplace=True)
data.head()

review_rating                                        review_text  \
0              5  As you get older, you know what you like and w...   
1              5  Three gigantic marmite jars that will last pro...   
2              4                                          Excellent   
3              5         A great flavour top - up for slow cooking.   
4              5                          Does what is says it does   

   verified_purchase  
0               True  
1               True  
2               True  
3               True  
4              False

>Titles are actually not mandatory in amazon reviews, and hence there are multiple missing values within the review_title. For this project, we are not going to be utlizing this dataset, and hence it is going to be dropped.

In [74]:
#SAVING UPDATED DATAFRAME AS .csv FILE
data.to_csv('C:/Data_Science/Mini Project/Capston Project/Capston_IOD_Project/DATABASE files/updated_data.csv')